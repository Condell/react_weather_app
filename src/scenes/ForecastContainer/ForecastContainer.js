import React, { Component } from "react";
import PropTypes from "prop-types";
import queryString from "query-string";
import { Redirect } from "react-router-dom";

import { fetch5DayForecast } from "../../services/api";
import Forecast from "./components/Forecast/Forecast";

class ForecastContainer extends Component {
  state = {
    weatherData: {},
    loading: true
  };

  handleClick = (day, name) => {
      this.props.history.push({
      pathname: `/detail/${name}`,
      state: day
    });
  }

  componentDidMount() {
    this.city = queryString.parse(this.props.location.search).city;
    fetch5DayForecast(this.city).then(res => {
      this.setState({
        weatherData: res,
        loading: false
      });
    });
  }

  componentWillReceiveProps(nextProps) {
    this.city = queryString.parse(nextProps.location.search).city;
    this.setState({
      loading: true
    });
    fetch5DayForecast(this.city).then(res => {
      this.setState({
        weatherData: res,
        loading: false
      });
    });
  }

  render() {
    return (
      <Forecast
        isLoading={this.state.loading}
        weatherData={this.state.weatherData}
        city={this.city}
        handleClick={this.handleClick}
      />
    );
  }
}

export default ForecastContainer;
