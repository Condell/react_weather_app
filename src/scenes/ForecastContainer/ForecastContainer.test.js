import React from "react";
import { shallow } from "enzyme";
import 'jest-styled-components'
import toJson from "enzyme-to-json"

import ForecastContainer from "./ForecastContainer";

jest.mock("./ForecastContainer.js", () => "ForecastContainer")

describe("ForecastContainer", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<ForecastContainer />);
  });

  it("should render", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});