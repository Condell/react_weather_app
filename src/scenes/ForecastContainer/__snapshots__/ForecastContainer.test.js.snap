// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastContainer should render 1`] = `
<Forecast
  handleClick={[Function]}
  isLoading={true}
  weatherData={Object {}}
/>
`;
